analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/RestrictDeserializedTypes.LosFormatter.vb

```diff
@@ -0,0 +1,87 @@
+        Else
+            Dim formatter3 = New LosFormatter(condition, "")                                    ' Noncompliant
+        End If
+    End Sub
+
+    Public Sub LocalVariables()
+        Dim trueVar = True
+        Dim formatter1 = New LosFormatter(trueVar, "")                                          ' Compliant
+        Dim falseVar = False
+        Dim formatter2 = New LosFormatter(falseVar, "")                                         ' Noncompliant
+    End Sub
+
+    Public Sub TernaryOp(condition As Boolean)
+        Dim falseVar = If(condition, False, False)
+        Dim formatter1 = New LosFormatter(falseVar, "")                                         ' Noncompliant
+        Dim trueVar = If(condition, True, True)
+        Dim formatter2 = New LosFormatter(trueVar, "")                                          ' Compliant
+        Dim formatter3 = New LosFormatter(If(condition, False, True), "")                       ' Noncompliant
+    End Sub
+
+    Public Function UnsafeReturnValue() As LosFormatter
+        Return New LosFormatter(False, "")                                                      ' Noncompliant
+    End Function
+
+    Public Function SafeReturnValue() As LosFormatter
+        Return New LosFormatter(True, "")
+    End Function
+
+    Public Sub InLambdaFunction()
+        Dim createSafe As Func(Of LosFormatter) = Function() New LosFormatter(True, "")         ' Compliant
+        Dim createUnsafe As Func(Of LosFormatter) = Function() New LosFormatter(False, "")      ' Noncompliant
+    End Sub
+
+    Public Function Switch(condition As Boolean) As LosFormatter
+        Select Case condition
+            Case True
+                Return New LosFormatter(condition, "")                                          ' Noncompliant - FP: engine does not learn from switch branches
+            Case Else
+                Return New LosFormatter(condition, "")                                          ' Noncompliant
+        End Select
+    End Function
+
+    Public Sub DataFlow()
+        Dim condition = False
+        condition = True
+
+        If condition Then
+            Dim formatter1 = New LosFormatter(condition, "")
+        Else
+            Dim formatter2 = New LosFormatter(condition, "")                                    ' Unreachable
+        End If
+
+        condition = False
+
+        If condition Then
+            Dim formatter3 = New LosFormatter(condition, "")                                    ' Unreachable
+        Else
+            Dim formatter4 = New LosFormatter(condition, "")                                    ' Noncompliant
+        End If
+
+        Dim formatter5 = New LosFormatter(New Boolean(), "")                                    ' Noncompliant
+        Dim formatter6 = New LosFormatter(Nothing, "")                                          ' Noncompliant FP - engine does not learn bool default value
```

Comment:
This is a TP. `Nothing` defaults to `false`.
This and the line above are wrongly annotated in the C# tests as well.